mod/display.php

```diff
@@ -379,114 +342,81 @@ function display_content(App $a, $update = 0) {
 		$item_id
 	);
 
-	if (!DBM::is_result($r) && local_user()) {
-		// Check if this is another person's link to a post that we have
-		$r = dba::fetch_first("SELECT `item`.uri FROM `item`
-			WHERE (`item`.`id` = ? OR `item`.`uri` = ?)
-			LIMIT 1",
-			$item_id,
-			$item_id
-		);
-		if (DBM::is_result($r)) {
-			$item_uri = $r['uri'];
-
-			$r = dba::p(item_query()." AND `item`.`uid` = ?
-				AND `item`.`parent` = (SELECT `parent` FROM `item` WHERE `uri` = ? AND uid = ?)
-				ORDER BY `parent` DESC, `gravity` ASC, `id` ASC",
-				local_user(),
-				$item_uri,
-				local_user()
-			);
-		}
+	if (!DBM::is_result($r)) {
+		notice(t('Item not found.') . EOL);
+		return $o;
 	}
 
-	if (DBM::is_result($r)) {
-		$s = dba::inArray($r);
+	$s = dba::inArray($r);
 
-		if ((local_user()) && (local_user() == $a->profile['uid'])) {
-			$unseen = dba::select('item', array('id'), array('parent' => $s[0]['parent'], 'unseen' => true), array('limit' => 1));
-			if (DBM::is_result($unseen)) {
-				dba::update('item', array('unseen' => false), array('parent' => $s[0]['parent'], 'unseen' => true));
-			}
+	if (local_user() && (local_user() == $a->profile['uid'])) {
+		$unseen = dba::select('item', array('id'), array('parent' => $s[0]['parent'], 'unseen' => true), array('limit' => 1));
+		if (DBM::is_result($unseen)) {
+			dba::update('item', array('unseen' => false), array('parent' => $s[0]['parent'], 'unseen' => true));
 		}
+	}
 
-		$items = conv_sort($s, "`commented`");
-
-		if (!$update) {
-			$o .= "<script> var netargs = '?f=&nick=" . $nick . "&item_id=" . $item_id . "'; </script>";
-		}
-		$o .= conversation($a, $items, 'display', $update);
-
-		// Preparing the meta header
-		require_once('include/bbcode.php');
-		require_once("include/html2plain.php");
-		$description = trim(html2plain(bbcode($s[0]["body"], false, false), 0, true));
-		$title = trim(html2plain(bbcode($s[0]["title"], false, false), 0, true));
-		$author_name = $s[0]["author-name"];
-
-		$image = $a->remove_baseurl($s[0]["author-thumb"]);
+	$items = conv_sort($s, "`commented`");
 
-		if ($title == "") {
-			$title = $author_name;
-		}
+	if (!$update) {
+		$o .= "<script> var netargs = '?f=&item_id=" . $item_id . "'; </script>";
+	}
+	$o .= conversation($a, $items, 'display', $update_uid);
 
-		// Limit the description to 160 characters
-		if (strlen($description) > 160) {
-			$description = substr($description, 0, 157) . '...';
-		}
+	// Preparing the meta header
+	require_once('include/bbcode.php');
+	require_once("include/html2plain.php");
```

Comment:
Code standards: Please remove parentheses from language constructs and please choose a single quote style, preferably single.